src/actions/search.js

```diff
@@ -82,6 +85,89 @@ export function clearSearch() {
     };
 }
 
+export function getFlaggedPosts() {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const userId = getCurrentUserId(state);
+        const teamId = getCurrentTeamId(state);
+
+        dispatch({type: SearchTypes.SEARCH_FLAGGED_POSTS_REQUEST}, getState);
+
+        let posts;
+        try {
+            posts = await Client4.getFlaggedPosts(userId, '', teamId);
+            await Promise.all([
+                getProfilesAndStatusesForPosts(posts.posts, dispatch, getState),
+                getMissingChannelsFromPosts(posts.posts)(dispatch, getState),
+            ]);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(batchActions([
+                {type: SearchTypes.SEARCH_FLAGGED_POSTS_FAILURE, error},
+                logError(error)(dispatch),
+            ]), getState);
+            return {error};
+        }
+
+        dispatch(batchActions([
+            {
+                type: SearchTypes.RECEIVED_SEARCH_FLAGGED_POSTS,
+                data: posts,
+            },
+            {
+                type: SearchTypes.SEARCH_FLAGGED_POSTS_SUCCESS,
+            },
+        ], 'SEARCH_FLAGGED_POSTS_BATCH'), getState);
+
+        return {data: posts};
+    };
+}
+
+export function getRecentMentions() {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const teamId = getCurrentTeamId(state);
+
+        dispatch({type: SearchTypes.SEARCH_RECENT_MENTIONS_REQUEST}, getState);
+
+        let posts;
+        try {
+            const termKeys = getCurrentUserMentionKeys(state).filter(({key}) => {
+                return key !== '@channel' && key !== '@all' && key !== '@here';
+            });
+
+            const terms = termKeys.map(({key}) => key).join(' ').trim() + ' ';
```

Comment:
Not something that would block this PR, but why are we adding a space to the end of the search terms?

Reply:
Copied from the webapp, so I guess there is a reason for it